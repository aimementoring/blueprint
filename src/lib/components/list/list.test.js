import React from 'react';
import renderer from 'react-test-renderer';
import List from './list.js';

describe('List', () => {
  it('renders properly', () => {
    const tree = renderer.create(<List>Gradient default List</List>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('is backwards compatible (text passed as prop)', () => {
    const tree = renderer
      .create(<List text="Gradient default List" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('works with children (even without a WWCC!)', () => {
    const tree = renderer
      .create(
        <List>
          <span>1</span>
          <span>2</span>
          <span>3</span>
        </List>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
