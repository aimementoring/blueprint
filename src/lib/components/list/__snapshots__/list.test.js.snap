// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List is backwards compatible (text passed as prop) 1`] = `
<div
  className="theme-base"
>
  <ul
    className="ulList"
  />
</div>
`;

exports[`List renders properly 1`] = `
<div
  className="theme-base"
>
  <ul
    className="ulList"
  />
</div>
`;
